generatebundlefile/ecr.go

```diff
@@ -206,14 +206,33 @@ func (d DockerAuthFile) Remove() error {
 }
 
 // getNameAndVersionPrivate looks up the latest pushed helm chart's tag from a given repo name from ECR.
-func (c *SDKClients) getNameAndVersion(s, accountID string) (string, string, string, error) {
+func (c *SDKClients) getNameAndVersion(repoName, tag, accountID string) (string, string, string, error) {
 	var version string
 	var sha string
-	splitname := strings.Split(s, ":") // TODO add a regex filter
+	splitname := strings.Split(repoName, ":") // TODO add a regex filter
 	name := splitname[0]
-	if len(splitname) == 1 {
+	ecrname := fmt.Sprintf("%s.dkr.ecr.us-west-2.amazonaws.com/%s", accountID, name)
+	if len(splitname) > 0 {
+		// If for promotion, we use a named tag instead of latest we do a lookup for that tag.
+		if !strings.Contains(tag, "latest") {
+			var imagelookup []ecrtypes.ImageIdentifier
+			imagelookup = append(imagelookup, ecrtypes.ImageIdentifier{ImageTag: &tag})
+			ImageDetails, err := c.ecrClient.Describe(&ecr.DescribeImagesInput{
+				RepositoryName: aws.String(repoName),
+				ImageIds:       imagelookup,
+			})
+			if err != nil {
+				return "", "", "", fmt.Errorf("error: Unable to complete DescribeImagesRequest to ECR. %s", err)
+			}
+			for _, images := range ImageDetails {
+				if len(images.ImageTags) == 1 {
+					return ecrname, tag, *images.ImageDigest, nil
+				}
+			}
+		}
+		// If for promotion we use latest tag we do a time based lookup to find the latest.
 		ImageDetails, err := c.ecrClient.Describe(&ecr.DescribeImagesInput{
-			RepositoryName: aws.String(s),
+			RepositoryName: aws.String(repoName),
```

Comment:
This name change is a nice improvement. 😄

Reply:
haha yea